packages/start/src/config/index.ts

```diff
@@ -219,6 +223,27 @@ export function defineConfig(
         dir: './public',
         base: '/',
       },
+      {
+        name: 'api',
+        type: 'http',
+        target: 'server',
+        base: '/api',
+        handler: importToProjectRelative('@tanstack/start/server-handler/api'),
+        routes: (router, app) =>
+          new TanStackStartServerFileSystemRouter(
+            { dir: tsrConfig.routesDirectory, extensions: FILE_EXTENSIONS },
+            router,
+            app,
+          ),
+        plugins: () => [
```

Comment:
I think this is the next step to make sure the correct handler is imported regardless of `dev` or `prod`.